src/celeritas/user/detail/SDParams.cc

```diff
@@ -0,0 +1,102 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/user/detail/SDParams.cc
+//---------------------------------------------------------------------------//
+#include "SDParams.hh"
+
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/io/Join.hh"
+#include "geocel/GeoVolumeFinder.hh"
+
+namespace celeritas
+{
+
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from list of volume labels.
+ */
+SDParams::SDParams(std::string output_label,
+                   VecLabel labels,
+                   GeoParamsInterface const& geo)
+    : output_label_{std::move(output_label)}, volume_labels_{std::move(labels)}
+{
+    CELER_EXPECT(!output_label_.empty());
+    if (volume_labels_.empty())
+    {
+        return;
+    }
+    CELER_EXPECT(!volume_labels_.empty());
```

Comment:
Let's require that the user provide at least one volume with this constructor. If the problem has no SDs, we could just use the default constructor.

Reply:
Default constructor added and is now used in optical `CoreParams` when no detector_labels are included in `CoreParams::input`